### ISSUE TITLE
Ability to change faction and leader after initially choice / Make choice private (GNA #13512)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1160]
**[Original submission](https://gna.org/bugs/index.php?13512)** by ancestral on 2009-05-12

#1. For multiplayer games, players who are connecting to a host must choose a faction and a leader. After making your selection, if you change your mind and want a different one, then you have two options:

1. Ask the host nicely to change it for you. (Hopefully they will.)

2. Drop out and join the game again.

Neither is a very good solution, as #1 involves extra work for the host, assuming the host has read the player's plea and the player knows exactly the unit s/he wants, and #2 forces the player to leave, which might send the wrong signal to the other players who might decide to abandon the game, or on the contrary, the game could fill up with a new player while the old player leaves and attempts to rejoin.

So, why not allow the player to change his/her choice after the initial selection, before the game starts?

#2. When choosing a side, players can try to choose a faction in response to another doing so. If a player sees there are a lot of Rebels, he may choose to be Drakes, etc. This can be a benefit, to add variety to the game, but it can also be quite manipulative, as people can choose their faction based on others' selections, potentially exploiting weaknesses.

The only workaround to not letting someone know who you're choosing is to choose Random, but then you too don't know who you're going to be. For example, a newer player might be good with Loyalists but not very good with any other faction woul be at an even greater disadvantage if they end up with, say, the Undead.

A wonderful idea would be to make your selection private. People don't need to know who you're playing anyway, and if players want to know then they can ask/tell them in the pre-game chat.

If idea #1 is adopted, then you could make it private to the host as well, making the playing field completely free from any manipulation at all.

(Reproduced on Mac OS X 10.5.6)
Release: 1.6.1
Priority: 5 - Normal
Severity: 1 - Wish

**Comment by wesnoth-bugs** [comment]
**Modified on 2009-05-13**

**em3 wrote:**

> Shouldn't it be two separate feature requests?
> 
> On a side note: I think both features should be optional and be enabled or disabled by host.

**Comment by Wedge009** [comment]
I don't know if things have changed, or maybe if difference MP scenarios have different rules, but I gave this a try in 1.14 and am able to change teams more than once after joining. So I suppose this only leaves the request to have an option for private faction selection.

**Comment by gfgtdf** [comment]
> I don't know if things have changed, or maybe if difference MP scenarios have different rules, but I gave this a try in 1.14 and am able to change teams more than once after joining

Yes this is new, iirc i added it on 1.14.4 so it was not possible in 1.14.0 or earlier.
